networksession.go

```diff
@@ -92,6 +96,29 @@ func newSessionUDP(connection Conn, srv *Server, sessionUDPData *SessionUDPData)
 		sessionUDPData: sessionUDPData,
 		mapPairs:       make(map[[MaxTokenSize]byte]map[uint16](*sessionResp)),
 	}
+
+	// set up noise initiator or receiver
+	cs := noise.NewCipherSuite(noise.DH25519, noise.CipherAESGCM, noise.HashSHA512)
+	rng := new(RandomInc)
+
+	hs, _ := noise.NewHandshakeState(noise.Config{
+		CipherSuite: cs,
+		Random:      rng,
+		Pattern:     noise.HandshakeNN,
+		Initiator:   initiator,
+	})
+
+	//log.Printf("newSessionUDP %p with HS %p", s, hs)
+	//debug.PrintStack()
+
+	s.ns = &NoiseState{
+		Hs:			hs,
+		Handshakes: 0,
+		Cs0:		nil,
+		Cs1:		nil,
```

Comment:
No need to declar Cs0, Handshakes, etc, since they'll default to the null value (i think)

Reply:
Indeed. When creating an instance of a struct, Go sets all omitted fields to their zero value (which is `nil` for references). Here, `Handshakes` could also be omitted (unless you still want in here for clarity).